### PR TITLE
feat: add radio generator

### PR DESCRIPTION
**Pull request by kedevked**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ng-primitives/ng-primitives/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation changes
- [ ] Other... Please describe:

## What does this PR implement/fix?

Add a generator for a radio component

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by kedevked** [comment]
I did base the generated code upon this: https://stackblitz.com/edit/angular-es2sm2-1x3uku?file=src%2Fmain.ts

**Comment by ashley-hunter** [comment]
Thank you! I had some time to put together an example of how I expect the components to be used in an app. Hopefully this is of some help:

https://stackblitz.com/edit/stackblitz-starters-vjuspo?file=src%2Fmain.ts

**Review by ashley-hunter — CHANGES_REQUESTED** [review]
Looks good, just had a suggestion for the output components

**Comment by kedevked** [comment]
The generator generates a radio-item component as well now

File: packages/ng-primitives/schematics/radio/generator.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+:host[data-checked="true"] [ngpRadioIndicator] {}
+
+.indicator-dot {}
+
+:host[data-checked="true"] .indicator-dot {}
+
+p {}
+
+:host[data-checked="true"] p {}`;
```
Comment (ashley-hunter) [inline packages/ng-primitives/schematics/radio/generator.ts]: I think we can remove a few of these rules, we don't know that there would be a `p` tag or an `.indicator-dot` etc.., so we can remove these and keep only the things that are a "given"

```suggestion
:host {}

:host[data-hover="true"] {}

:host[data-focus-visible="true"] {}

:host[data-press="true"] {}

:host[data-checked="true"] {}

[ngpRadioIndicator] {}

:host[data-checked="true"] [ngpRadioIndicator] {}
`;
```

**Review by ashley-hunter — CHANGES_REQUESTED** [review]
Looks good, just one small suggestion

**Review by ashley-hunter — APPROVED** [review]
All good, thank you

**Pull request merged**